Unix/tests/cli/test_cli.cpp

```diff
@@ -337,15 +407,57 @@ static int StopServerSudo()
     if (Process_StopChild(&serverProcess) != 0)
         return -1;
 #else
-    if (!skipTest)
+    if (serverStarted)
     {
-        std::stringstream cmd;
-        cmd << "sudo kill -15 " << (int)serverProcess.reserved;
-        if (system(cmd.str().c_str()) == -1)
+        uint args = 0;
+        const char* argv[MAX_SERVER_ARGS];
+        std::stringstream pidStr;
+        int pid;
+        Process killProcess;
+        int status;
+        int numWaits = 0;
+
+        if (PIDFile_Read(&pid) != 0)
+            return -1;
+
+        pidStr << pid;
+
+        argv[args++] = sudoPath;
+        argv[args++] = "kill";
+        argv[args++] = "-15";       // sudo kill -s SIGTERM doesn't work on Sun, a bug?
+        argv[args++] = pidStr.str().c_str();
+        argv[args++] = NULL;
+
+        if (Process_StartChild(&killProcess, sudoPath, (char**)argv) != 0)
             return -1;
+        
+try_again:
+        /* Wait for original process to go */
+        pid_t p = waitpid(pid, &status, 0);
+        if (p == -1)
+        {
+            if (errno == ECHILD)
+                goto cleanup;
+
+            // EINTR is common in waitpid() and doesn't necessarily indicate failure.
+            //       Give the system 10 times to see if the pid cleanly exits.
+            if (10 < numWaits && errno == EINTR)
+            {
+                numWaits++;
+                Sleep_Milliseconds(10);
```

Comment:
Is 10 milliseconds (really 10*10 milliseconds, or 100 milliseconds) enough, even on our really slow HP systems? Perhaps, if UNIX, make it longer, like 25 tries (1/2 second)?

If the process finishes quickly, this change will have no effect. It'll only matter if the system is slow and takes longer than you anticipated ...